pennylane/labs/trotter_error/product_formulas/bch.py

```diff
@@ -0,0 +1,435 @@
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""This file contains the BCH computation"""
+
+from __future__ import annotations
+
+import math
+from collections import defaultdict
+from collections.abc import Hashable
+from itertools import permutations, product
+from typing import TYPE_CHECKING, Dict, Generator, List, Sequence, Tuple
+
+import numpy as np
+
+if TYPE_CHECKING:
+    from pennylane.labs.trotter_error import ProductFormula
+
+
+def bch_expansion(
+    product_formula: ProductFormula, order: int
+) -> List[Dict[Tuple[Hashable], complex]]:
+    r"""Compute the Baker-Campbell-Hausdorff expansion of a :class:`~.pennylane.labs.trotter_error.ProductFormula` object.
+
+    Args:
+        product_formula (:class:`~.pennylane.labs.trotter_error.ProductFormula`): A :class:`~.pennylane.labs.trotter_error.ProductFormula` object.
+        order (int) The maximum order of the expansion to return.
+    Returns:
+        List[Dict[Tuple[Hashable], complex]]: A list of dictionaries. The ``ith`` dictionary contains the ``ith`` order commutators and their coefficients.
```

Comment:
Please briefly explain this, maybe in the example. Not sure the it is clear what this dictionary represents.  

```
defaultdict(<class 'complex'>,
             {('A',): (1+0j),
              ('B',): (1+0j),
              ('C',): (1+0j)}),
```